### ISSUE TITLE
Throw error on attempt to link property that has already been linked

### ISSUE THREAD

**Issue by Lamby777** [Lamby777/dialogical#2]
tl;dr: Make the compiler complain if you do something like this:
```dg
% Link Twice
###

Link NAME Mira
VOX Mira

Link NAME Dylan
VOX Dylan
VOX Ethan
// ^^^ LINE ABOVE CAUSES COMPTIME ERROR

###
---
NAME Mira

wuzzup fools

---

```

Right now, I have no idea what it does in this situation and I don't care to find out because it's not even a valid state for the compiler to be in. It should just throw an error and exit if it ever encounters a `Link` command with an already existing key in its following lines.

Or maybe I could make it just log a warning and continue? I don't think that would be a good idea because then people could just ignore the error and continue writing shitty code...

**Comment by Lamby777** [comment]
This would also let us free up the requirement for specifying the value to unlink in the lines following an `Unlink` directive... which is pretty much a required feature before I can even consider that command "complete."

Instead of

```
Unlink NAME Mira
VOX Mira
```

you can then just do

```
Unlink NAME Mira
VOX
```

with a little bit of tweaking to how the `Unlink` command works

**Comment by Lamby777** [comment]
if this gets finished, we're basically in 1.0 release

no real promises on the version bump tho since i like making breaking changes :3

**Comment by Lamby777** [comment]
update: should also error if a link refers to its own property, like
```
Link NAME Cherry
NAME Walter White
```
should not compile... and
```
Link NAME Cherry
NAME Walter White
NAME Your Mom
```
DEFINITELY should not compile (breaks both rules).

**Comment by Lamby777** [comment]
actually, nah. i've reconsidered that second point. it might end up being a footgun, but we'll see. it might just be useful to avoid repeating yourself, when you can do smt like
```
Link NAME John
NAME John James Jingleton III
```

it'd basically be the `dg` equivalent of compile-time constants.

**Comment by Lamby777** [comment]
Probably needs to be a separate thread cuz this one is becoming a sort of "mega-issue" but if we're gonna allow that ^^^ stuff then there needs to also be a compiler error for when a cycle is created, like
```
Link NAME Mira
NAME Agent Cascade

Link NAME Agent Cascade
NAME Mira
```

**Comment by Lamby777** [comment]
Fixed in 0.5.2
